src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -385,6 +385,7 @@ internal List<LibraryDependency> GetSpecDependencies(
 
                 for (var i = 0; i < dependencies.Count; i++)
                 {
+                    bool isPruned = IsDependencyPruned(dependencies[i], targetFrameworkInfo.PackagesToPrune);
```

Comment:
I recommend moving this line to after we declare the variable `dependency` so we don't have to dereference the dependencies at index `i` twice, just once, and then operate on the same result. It adds clarity and reduces the chance of introducing a bug in the future.